src/main/java/org/folio/fqm/service/EntityTypeFlatteningService.java

```diff
@@ -279,6 +283,7 @@ private static EntityTypeSource copySource(EntityTypeSource sourceFromParent, En
         .condition(source.getJoin().getCondition())
         .joinTo(renameAliases ? renamedAliases.get(source.getJoin().getJoinTo()) : source.getJoin().getJoinTo())
       )
+      .essentialOnly(sourceFromParent == null || Boolean.TRUE.equals(source.getEssentialOnly()))
```

Comment:
I think I'd just be lazy on this one and copy the value blindly, rather than force it to be non-null
```suggestion
      .essentialOnly(source.getEssentialOnly())
```